src/Tizen.Nlp/Tizen.Nlp/NaturalLanguageProcess.cs

```diff
@@ -0,0 +1,370 @@
+                {
+                    _mapsNamedEntity[requestid]?.Invoke(e);
+                    _mapsNamedEntity.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("lemmatize"))
+            {
+                requestid = int.Parse((string)msg.GetItem("request_id"));
+                e.Message = result;
+                if (_mapsLemmatize.ContainsKey(requestid))
+                {
+                    _mapsLemmatize[requestid]?.Invoke(e);
+                    _mapsLemmatize.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("langdetect"))
+            {
+                e.Message = result;
+                if (_mapsLangDetect.ContainsKey(requestid))
+                {
+                    _mapsLangDetect[requestid]?.Invoke(e);
+                    _mapsLangDetect.Remove(requestid);
+                }
+            }
+            else
+            {
+                return;
+            }
+
+            Log.Debug(_tag, "done");
+        }
+
+        /// <summary>
+        /// An construct method  to init local env of NLP .
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public NaturalLanguageProcess()
+        {
+            _tag = Application.Current.ApplicationInfo.ApplicationId;
+            Log.Debug(_tag, "msg construct started");
+            _msg = new Message(ServiceId);
+            _msg.Disconnected += (s, e) =>
+            {
+                _isConnected = false;
+                DisconnectEvent?.Invoke(e);
+            };
+            Log.Debug(_tag, "msg construct success");
+            _noti.Received += ResultReceived;
+            Log.Debug(_tag, "notify callback be assigned");
+            Log.Debug(_tag, "start to connect");
+            Log.Debug(_tag, "wait to callback of onConnected");
+        }
+
+
+        /// <summary>
+        /// An async method  to connect remote service.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        /// <exception cref="InvalidIDException">Thrown when the connect is rejected.</exception>
+        public Task Connect()
+        {
+            TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();
+            _msg.Connected += (sender, e) =>
+            {
+                Log.Debug(_tag, "start to register");
+                _msg.CoRegister(Application.Current.ApplicationInfo.ApplicationId, _noti);
+                Log.Debug(_tag, "connected callback be called");
+                tcs.SetResult(true);
+                _isConnected = true;
+            };
+            _msg.Rejected += (sender, e) => { tcs.SetException(new InvalidIDException()); };
+            _msg.Connect();
+            return tcs.Task;
+        }
+
+        private void Close()
+        {
+            _noti.Received -= ResultReceived;
+            _msg.UnRegister();
+            _msg.Dispose();
```

Comment:
If _msg need to be disposed when closed, 
_msg shoud be created in Connect()
We could re-connect after close

``` c#
nlp.Connect();
...
nlp.Close();
...
nlp.Connect();
...
nlp.Close();
```